src/Compilers/CSharp/Test/Syntax/Parsing/MemberDeclarationParsingTests.cs

```diff
@@ -12205,6 +12219,86 @@ namespace N
             EOF();
         }
 
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/74482")]
+        public void ExtraTypeOnlyMembers_DoNotPullNextClassInEvenIfPrivate()
+        {
+            // Test all the different type-only member forms.
+            var text = $$"""
+                namespace N
+                {
+                    class C
+                    {
+                    }
+
+                    // Will get pulled into C
+                    void Method()
+                    {
+                    }
+
+                    // Not currently pulled in.  But could consider it in the future.
+                    private class T
```

Comment:
Can you add another orphan member after the private class?

Reply:
sure.